lib/proj-parser.js

```diff
@@ -0,0 +1,77 @@
+var fs = require('fs');
+var path = require('path');
+var parseXML = require('xml2js').parseString;
+var debug = require('debug')('snyk');
+
+function determineDotnetVersion(rootDir) {
+  var csprojPath = findFile(rootDir, /.*\.csproj$/);
+  if (!csprojPath) {
+    throw new Error('.csproj file not found in ' + rootDir + '. ' +
+      'Make sure project was built before running `snyk test`');
+  }
+  debug('Checking .net framework version in .csproj file ' +
+    csprojPath);
+
+  var csprojContents = fs.readFileSync(csprojPath);
+
+  var frameworks = [];
+  parseXML(csprojContents, function (err, parsedCsprojContents) {
+    if (err) {
+      throw err;
+    }
+
+    var versionLoc = parsedCsprojContents.Project.PropertyGroup[0];
+
+    if (versionLoc.TargetFrameworkVersion) {
+      frameworks.push(toReadableVersion(versionLoc.TargetFrameworkVersion[0]));
+    } else if (versionLoc.TargetFramework) {
+      frameworks.push(toReadableVersion(versionLoc.TargetFramework[0]));
+    } else if (versionLoc.TargetFrameworks) {
+      versionLoc.TargetFrameworks[0].split(';').forEach(function (version) {
+        var frameworkVersion = toReadableVersion(version);
+        if (frameworkVersion) {
+          frameworks.push(frameworkVersion);
+        }
+      });
+    } else {
+      throw new Error('Could not find TargetFrameworkVersion/TargetFramework/' +
+        'TargetFrameworks defined in your .csproj file');
+    }
+  });
+  if (frameworks.length < 1) {
+    throw new Error('Could not find valid/supported .NET version');
+  }
+  return frameworks;
+}
+
+function toReadableVersion(version) {
+  var typeMapping = {
+    v: '.NETFramework',
+    net: '.NETFramework',
+    netstandard: '.NETStandard',
+    netcoreapp: '.NETCore',
+  };
+
+  for (var type in typeMapping) {
+    if (new RegExp(type + /\d.?\d(.?\d)?$/.source).test(version)) {
+      return typeMapping[type] + version.split(type)[1];
+    }
+  }
```

Comment:
what's the ret val in case there is no match?

Reply:
undefined. And there is a check later:
```
  if (frameworks.length < 1) {
    throw new Error('Could not find valid/supported .NET version');
  }
```